OMCompiler/SimulationRuntime/c/simulation/solver/sundials_util.c

```diff
@@ -0,0 +1,78 @@
+/*
+ * This file is part of OpenModelica.
+ *
+ * Copyright (c) 1998-2020, Open Source Modelica Consortium (OSMC),
+ * c/o Linköpings universitet, Department of Computer and Information Science,
+ * SE-58183 Linköping, Sweden.
+ *
+ * All rights reserved.
+ *
+ * THIS PROGRAM IS PROVIDED UNDER THE TERMS OF THE BSD NEW LICENSE OR THE
+ * GPL VERSION 3 LICENSE OR THE OSMC PUBLIC LICENSE (OSMC-PL) VERSION 1.2.
+ * ANY USE, REPRODUCTION OR DISTRIBUTION OF THIS PROGRAM CONSTITUTES
+ * RECIPIENT'S ACCEPTANCE OF THE OSMC PUBLIC LICENSE OR THE GPL VERSION 3,
+ * ACCORDING TO RECIPIENTS CHOICE.
+ *
+ * The OpenModelica software and the OSMC (Open Source Modelica Consortium)
+ * Public License (OSMC-PL) are obtained from OSMC, either from the above
+ * address, from the URLs: http://www.openmodelica.org or
+ * http://www.ida.liu.se/projects/OpenModelica, and in the OpenModelica
+ * distribution. GNU version 3 is obtained from:
+ * http://www.gnu.org/copyleft/gpl.html. The New BSD License is obtained from:
+ * http://www.opensource.org/licenses/BSD-3-Clause.
+ *
+ * This program is distributed WITHOUT ANY WARRANTY; without even the implied
+ * warranty of MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE, EXCEPT AS
+ * EXPRESSLY SET FORTH IN THE BY RECIPIENT SELECTED SUBSIDIARY LICENSE
+ * CONDITIONS OF OSMC-PL.
+ *
+ */
+
+/** \file sundials_util.c
+ */
+
+#include "util/omc_error.h"
+
+#include "sundials_util.h"
+#include <sundials/sundials_matrix.h>
+#include <sunmatrix/sunmatrix_sparse.h>
+#include <nvector/nvector_serial.h>
+
+
+#ifdef WITH_SUNDIALS
+
+#define UNUSED(x) (void)(x)   /* Surpress compiler warnings for unused function input */
+
+/**
+ * @brief Set element of sparse Sundials matrix.
+ *
+ * Jac(row, column) = val.
+ *
+ * @param row       Row of matrix element.
+ * @param column    Column of matrix element.
+ * @param nth       Sparsity pattern lead index.
+ * @param value     Value to set in position (i,j).
+ * @param Jac       Pointer to double array storing matrix.
+ * @param nRows     Number of rows of Jacobian matrix, unused.
+ */
+void setJacElementSundialsSparse(int row, int column, int nth, double value, void* Jac, int nRows)
+{
+  UNUSED(nRows);   /* Disables compiler warning */
```

Comment:
You can skip the name of the argument and specify the type only. It would not complain anymore :sunglasses:. 
```suggestion
void setJacElementSundialsSparse(int row, int column, int nth, double value, void* Jac, int /*nRows*/)
```

Reply:
Oh I didn't know this was possible. Thanks.

Reply:
Turns out my gcc version is complaining and that this is more of a C++ thing: https://stackoverflow.com/a/8776886/11041668
I think C23 would allow it, though.

Reply:
That is unfortunate. I never realized it was a C++ only thing. However, now that I think about it, I have only used it in C++ code.